llvm/samples/clang/pom.xml

```diff
@@ -16,6 +16,19 @@
         </dependency>
     </dependencies>
     <build>
-        <sourceDirectory>.</sourceDirectory>
+        <sourceDirectory>${project.basedir}</sourceDirectory>
+        <resources>
+            <resource>
+                <directory>${project.basedir}</directory>
+                <includes>
+                    <include>**/*.cc</include>
+                </includes>
+                <excludes>
+                    <exclude>**/*.java</exclude>
+                    <exclude>**/pom.xml</exclude>
+                    <exclude>target/**/*</exclude>
+                </excludes>
+            </resource>
+        </resources>
```

Comment:
Why do we need this?

Reply:
```xml
        <resources>
            <resource>
                <directory>${project.basedir}</directory>
            </resource>
        </resources>
```

allows to load _Maven_ resources from the very same directory where Java sources reside (note that this project doesn't follow the standard project layout of `src/main/{java,resources}`). In other words, sample1.cc can now be opened not only as an external file, but also as a classpath resource (via `Class.getResource()`/`Class.getResourceAsStream()`).

The rest just adds inclusion/exclusion filters, so that Java sources and compiled classes are not treated as resources:

```xml
                <includes>
                    <include>**/*.cc</include>
                </includes>
                <excludes>
                    <exclude>**/*.java</exclude>
                    <exclude>**/pom.xml</exclude>
                    <exclude>target/**/*</exclude>
                </excludes>
```

Now, we can [load `sample1.cc` as a _classpath resource_](https://github.com/bytedeco/javacpp-presets/pull/1522/files#diff-f55f5480bb711e90a230d9805d56cad95cab99215c438b08c75722169439c098R78):

```java
        final URL codeUrl = requireNonNull(ClangMemoryMgmtExample.class.getResource("/sample1.cc"));
```